### ISSUE TITLE
WebJob using asp.net 5 console application

### ISSUE THREAD

**Issue by mburumaxwell** [Azure/azure-webjobs-sdk#566]
Hi,
I have a problem when I try to use the webjobs sdk with a console application.
The code:
```cs
var jobconfig = new JobHostConfiguration("key here");
var host = new JobHost(jobconfig);
host.RunAndBlock();
```

It fails in the first line with the following error:
```
An exception of type 'System.IO.FileLoadException' occurred in Microsoft.Azure.WebJobs.Host.dll but was not handled in user code
Additional information: A strongly-named assembly is required. (Exception from HRESULT: 0x80131044)
```

Checking further I get this
```
A strongly-named assembly is required. (Exception from HRESULT: 0x80131044)
```

The Stack Trace is:
```
   at Microsoft.Azure.WebJobs.Host.Executors.DefaultStorageCredentialsValidator..ctor()
   at Microsoft.Azure.WebJobs.Host.Executors.DefaultStorageAccountProvider..ctor()
   at Microsoft.Azure.WebJobs.JobHostConfiguration..ctor(String dashboardAndStorageConnectionString)
   at TestWebJob.Program.Main(String[] args) in C:\Users\Me\Documents\TestWebJob\Program.cs:line 20
```

After reading through some issues, I realized I should sign the output. I tried two options as below but none worked on the Console App:
Option 1:
```json
"compilationOptions": {
    "keyFile": "TestWebJob.snk"
  }
```

Option 2:
```json
"compilationOptions": {
    "strongName": true
  }
```

Anyone have a solution or workaround?
Thanks in advance

**Comment by mathewc** [comment]
You don't have to sign your job assembly. This should "just work" - my colleague Ahmed has even tried this scenario end to end (see his blob post here: http://ahmelsayed.com/running-dnx-based-webjobs/). Try following his steps. Also what version of the webjobs SDK are you using? I assume you're pulling the package from nuget? If you're trying to build the SDK yourself locally, you may need to add some skip signing exclusions for our delay signed assemblies (see here: https://github.com/Azure/azure-webjobs-sdk/wiki/Development).

**Comment by mburumaxwell** [comment]
Thank you for your response. I was able to setup the basic webjob and it could run. However, if I reference a class library built in Dnx I get the same strong name error. I added strong name option in the class library but still it could not work.

**Comment by mathewc** [comment]
Well, you're in uncharted territory - we haven't tried these integration scenarios yet since Asp.Net 5 is so new. We verified that Dnx WebJobs using the WebJobs SDK work. I'm not sure what you're running into here. Do you have detailed repro steps or a repo containing the repro that we can check out?

**Comment by mburumaxwell** [comment]
Hello, yes I did so.

**Comment by mburumaxwell** [comment]
@mathewc I have created a repo here https://github.com/mburumaxwell/AspNet5WebJobTest please look at the commit history to see the steps to reproducing the error.
Thank you in advance

**Comment by mathewc** [comment]
I closed this issue previously because you've also logged this issue on the ASP.NET 5 repo. I think that is the best place for help on this one, since the issue seems to be in the DNX host which they own.

**Comment by mburumaxwell** [comment]
I realize the error here only appears when I add the storage library. Then
the JobHostConfiguration s unable to load from the library which I believe
is through reflection. That's why I thought it still remains relevant to
the WebJobs SDK.

On Friday, September 11, 2015, Mathew Charles <notifications@github.com>
wrote:

> I closed this issue previously because you've also logged this issue on
> the ASP.NET 5 repo. I think that is the best place for help on this one,
> since the issue seems to be in the DNX host which they own.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Azure/azure-webjobs-sdk/issues/566#issuecomment-139598568>
> .
>


-- 
Regards,
Maxwell Weru

**Comment by mburumaxwell** [comment]
I found out the problem. The webjobs sdk internally references version 4.0.1 of the WindowsAzure.Storage package. As a result I solved the issue by directly referencing the webjobs sdk and not the storage but getting it by extension.
The relevant commit is here:
https://github.com/mburumaxwell/AspNet5WebJobTest/commit/0be429cfef0013c50cb04547ae29c8d7d91912ba
After that change everything works fine now.
